mlir/lib/Dialect/XeGPU/IR/XeGPUDialect.cpp

```diff
@@ -209,6 +209,19 @@ LayoutAttr::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// XeGPU_RangeAttr
+//===----------------------------------------------------------------------===//
+
+LogicalResult
+RangeAttr::verify(llvm::function_ref<mlir::InFlightDiagnostic()> emitError,
+                  IntegerAttr startOfRange, IntegerAttr endOfRange) {
+  if (startOfRange.getInt() >= endOfRange.getInt())
+    return emitError() << "EndOfRange must be greater than StartOfRange";
```

Comment:
nit: in the error, I'd refer to values to their attribute names `end` and `start`
it should improve error readability